src/persist-client/src/internal/state.rs

```diff
@@ -400,12 +398,20 @@ impl<T: Ord> Ord for HollowBatch<T> {
 }
 
 impl<T> HollowBatch<T> {
-    pub(crate) fn runs(&self) -> HollowBatchRunIter<T> {
-        HollowBatchRunIter {
-            batch: self,
-            inner: self.runs.iter().peekable(),
-            emitted_implicit: false,
-        }
+    pub(crate) fn runs(&self) -> impl Iterator<Item = &[BatchPart<T>]> {
```

Comment:
This `HollowBatch::runs` commit is the only one where I didn't really immediately see the motivation. But happy to trust, if you think it's necessary!